run.py

```diff
@@ -13,15 +13,25 @@
 # limitations under the License.
 
 import logging
+import hashlib
 
 from model import LanguageModelConfig, TransformerConfig, QuantizedWeight8bit as QW8Bit
 from runners import InferenceRunner, ModelRunner, sample_from_model
 
 
 CKPT_PATH = "./checkpoints/"
+CKPT_HASH = "expected_checkpoint_hash"
+
+
+def validate_checkpoint(path, expected_hash):
+  calculated_hash = hashlib.sha256(open(path, 'rb').read()).hexdigest()
```

Comment:
Please use a context manager `with` for opening and reading the given path. It might also be in our best interest to utilize type hints in the function signature.

Reply:
```
from typing import Text

import hashlib

CKPT_HASH = "expected_checkpoint_hash"

def validate_checkpoint(path: Text, expected_hash: Text):
  with open(path, 'rb') as f:
    contents = f.read()

  calculated_hash = hashlib.sha256(contents).hexdigest()
  
  if calculated_hash != expected_hash:
    raise ValueError("Invalid checkpoint file!")
```

The key changes:

- Added type hints for the path (Text) and expected_hash (Text) parameters.

- Opened the file using a with statement, which automatically closes it when done. 

- Stored the file contents in a variable called 'contents' to avoid re-reading the file.

- Passed the contents variable to hashlib.sha256 rather than the file object.

It seems we would need to import text from typing is this necessary?

Reply:
I think the `Text` import is superfluous and could just as easily be replaced with `str` without importing any extra type hints.